tests/PersistentStorageTest.php

```diff
@@ -11,61 +11,109 @@
     }
 
     public function fromArrayProvider(): Generator
     {
+        yield 'empty' => [
+            [],
+            PersistentStorage::create(),
+        ];
+
         yield 'readyTrue' => [
             ['ready' => true],
-            true,
+            PersistentStorage::create()->setReady(true),
         ];
+
         yield 'readyFalse' => [
             ['ready' => false],
-            false,
+            PersistentStorage::create()->setReady(false),
         ];
+
         yield 'readyNull' => [
             ['ready' => null],
-            null,
+            PersistentStorage::create(),
         ];
+
+        yield 'with k8sStorageClass' => [
+            ['k8sStorageClassName' => 'storage-class'],
+            PersistentStorage::create()->setK8sStorageClassName('storage-class'),
+        ];
+    }
+
+    /** @dataProvider toArrayProvider */
+    public function testToArray(PersistentStorage $persistentStorage, array $expectedValue): void
+    {
+        self::assertSame($expectedValue, $persistentStorage->toArray());
     }
 
     public function toArrayProvider(): Generator
     {
+        yield 'empty' => [
+            PersistentStorage::create(),
+            [
+                'ready' => null,
+            ],
+        ];
+
         yield 'readyTrue' => [
-            (new PersistentStorage())->setReady(true),
-            ['ready' => true],
+            PersistentStorage::create()->setReady(true),
+            [
+                'ready' => true,
+            ],
         ];
+
         yield 'readyFalse' => [
-            (new PersistentStorage())->setReady(false),
-            ['ready' => false],
+            PersistentStorage::create()->setReady(false),
+            [
+                'ready' => false,
+            ],
         ];
+
         yield 'readyNull' => [
-            (new PersistentStorage())->setReady(null),
-            ['ready' => null],
+            PersistentStorage::create()->setReady(null),
+            [
+                'ready' => null,
+            ],
         ];
+
+        yield 'with k8sStorageClass' => [
+            PersistentStorage::create()->setK8sStorageClassName('storage-class'),
+            [
+                'ready' => null,
+                'k8sStorageClassName' => 'storage-class',
+            ],
+        ];
+    }
+
+    public function testSetIsReady(): void
+    {
+        $persistentStorage = new PersistentStorage();
+        self::assertNull($persistentStorage->isReady());
+
+        $persistentStorage->setReady(true);
+        self::assertTrue($persistentStorage->isReady());
+
+        $persistentStorage->setReady(false);
+        self::assertFalse($persistentStorage->isReady());
+
+        $persistentStorage->setReady(null);
+        self::assertNull($persistentStorage->isReady());
+    }
+
+    public function testSetK8sStorageClassName(): void
+    {
+        $persistentStorage = new PersistentStorage();
+        self::assertSame('', $persistentStorage->getK8sStorageClassName());
```

Comment:
it's not null?

Reply:
It's the same old :hankey: as in the Project class - all string values are empty string by default so it can distinguish between "value not changed" (empty string) and value not set/value cleared (null).

Reply:
agh, yeah, I remember, not my favorite thing in the world for sure